deckhand/tests/unit/engine/test_document_layering.py

```diff
@@ -0,0 +1,392 @@
+
+import os
+import yaml
+
+import testtools
+
+from deckhand.engine import layering
+from deckhand import errors
+
+
+class TestDocumentLayering(testtools.TestCase):
+
+    FAKE_YAML_DATA_2_LAYERS = """
+---
+schema: deckhand/LayeringPolicy/v1
+metadata:
+  schema: metadata/Control/v1
+  name: layering-policy
+data:
+  layerOrder:
+    - global
+    - site
+---
+schema: example/Kind/v1
+metadata:
+  schema: metadata/Document/v1
+  name: global-1234
+  labels:
+    key1: value1
+  layeringDefinition:
+    abstract: true
+    layer: global
+data:
+  a:
+    x: 1
+    y: 2
+  c: 9
+---
+schema: example/Kind/v1
+metadata:
+  schema: metadata/Document/v1
+  name: site-1234
+  layeringDefinition:
+    layer: site
+    parentSelector:
+      key1: value1
+    actions:
+      - method: {method}
+        path: {path}
+data:
+  a:
+    x: 7
+    z: 3
+  b: 4"""
+
+    FAKE_YAML_DATA_3_LAYERS = """
+---
+schema: deckhand/LayeringPolicy/v1
+metadata:
+  schema: metadata/Control/v1
+  name: layering-policy
+data:
+  layerOrder:
+    - global
+    - region
+    - site
+---
+schema: example/Kind/v1
+metadata:
+  schema: metadata/Document/v1
+  name: global-1234
+  labels:
+    key1: value1
+  layeringDefinition:
+    abstract: true
+    layer: global
+data:
+  a:
+    x: 1
+    y: 2
+---
+schema: example/Kind/v1
+metadata:
+  schema: metadata/Document/v1
+  name: region-1234
+  labels:
+    key1: value1
+  layeringDefinition:
+    abstract: true
+    layer: region
+    parentSelector:
+      key1: value1
+    actions:
+      - method: replace
+        path: .a
+data:
+  a:
+    z: 3
+---
+schema: example/Kind/v1
+metadata:
+  schema: metadata/Document/v1
+  name: site-1234
+  layeringDefinition:
+    layer: site
+    parentSelector:
+      key1: value1
+    actions:
+      - method: merge
+        path: .
+data:
+  b: 4"""
+
+    def _read_data(self, filename):
+        dir_path = os.path.dirname(os.path.realpath(__file__))
+        test_yaml_path = os.path.abspath(os.path.join(
+            dir_path, os.pardir, 'resources', filename))
+
+        with open(test_yaml_path, 'r') as yaml_file:
+            yaml_data = yaml_file.read()
+        return self._parse_data(yaml_data)
+
+    def _parse_data(self, yaml_data):
+        return [d for d in yaml.safe_load_all(yaml_data)]
+
+    def _test_layering(self, expected, documents):
+        document_layering = layering.DocumentLayering(documents)
+        rendered_data = document_layering.render()
+        self.assertEqual(expected, rendered_data)
+
+
+class TestDocumentLayering2Layers(TestDocumentLayering):
+
+    def test_layering_method_delete(self):
+        expected = [{}, {'b': 4}]
+
+        for idx, path in enumerate(['.', '.a']):
+            documents = self._parse_data(self.FAKE_YAML_DATA_2_LAYERS.format(
+                method='delete', path=path))
+            self._test_layering(expected[idx], documents)
+
+    def test_layering_method_merge(self):
+        expected = [
```

Comment:
@mark-burnett  If you're doing a merge operation, with `path: .a`, why would you only include what's in path `.a` in the rendered result?

I rewrote my unit tests to make it easier to keep track of changes, so for:

```python

            kwargs = {
                "_GLOBAL_DATA_": {"data": {"a": {"x": 1, "y": 2}, "c": 9}},
                "_SITE_DATA_": {"data": {"a": {"x": 7, "z": 3}, "b": 4}},
                "_SITE_ACTIONS_": {
                    "actions": [{"method": "merge", "path": ".a"}]}
            }
```

Does this translate to:

1) Keep all the site data but merge the site data in path ".a" with the global data in path ".a". Or:
2) Only keep the result of merging the site data in path ".a" with the global data in path ".a".

2) Is not self-evident because "merge" implies "this plus this" versus "this plus this...except this".

Want to clarify this before making changes everywhere.